web/src/click_handlers.js

```diff
@@ -774,7 +774,9 @@ export function initialize() {
         e.preventDefault();
         e.stopPropagation();
 
-        window.location.hash = "narrow/is/dm";
+        if (!$(e.target).hasClass("direct-messages-list-filter")) {
+            window.location.hash = "narrow/is/dm";
+        }
```

Comment:
So in theory, we should have a very precise selector on the filter button, and the click handler for that should take precedence over this one and call `stopPropagation` to avoid this outer-container handler running, and thus you shouldn't need to have a check like this one -- which are always a code smell. Is there something I'm missing.

Reply:
Hmm, I guess the challenge is the element is a text input, not a button? How do we handle this for the similar topic list?

Reply:
I think this should probably get refactored in a follow-up PR; the logic seems to work OK, but it's fragile. And cleaning this up might require reworking the HTML structure a bit to give this a more accurate selector to latch onto.

So I'm going to let this remain when merging, and perhaps you can propose something in a follow-up PR?